app/assess/data.py

```diff
@@ -513,21 +513,62 @@ def get_file_response(file_name: str, application_id: str):
         mimetype = obj["ResponseMetadata"]["HTTPHeaders"]["content-type"]
         data = obj["Body"].read()
 
-        response = Response(
-            data,
-            mimetype=mimetype,
-            headers={
-                "Content-Disposition": (
-                    f"attachment;filename={quote_plus(file_name)}"
-                )
-            },
-        )
-        return response
+        return data, mimetype
     except ClientError as e:
         current_app.logger.error(e)
         raise Exception(e)
 
 
+def get_file_names_for_application_upload_fields(
+    application_id: str, short_id: str
+) -> List[tuple]:
+    """
+    This function retrieves the file names and download links
+    for a specific application in an S3 bucket.
+    The function filters the files based on the file metadata
+    'componentname' (file upload field keys).
+
+    Parameters:
+    application_id (str): The unique identifier of the application.
+
+    Returns:
+    List[tuple]: A list of tuples, where each tuple contains the
+    file name and its download link.
+    """
+    s3 = resource(
+        "s3",
+        aws_access_key_id=Config.AWS_ACCESS_KEY_ID,
+        aws_secret_access_key=Config.AWS_SECRET_ACCESS_KEY,
+        region_name=Config.AWS_REGION,
+    )
+    bucket = s3.Bucket(Config.AWS_BUCKET_NAME)
+    file_upload_field_key_list = ["ArVrka", "rFXeZo", "EEBFao", "ozgwXq"]
```

Comment:
This will only work for COF. Can we make it more generic, i.e. filter the json to find `file` type field answers?

Reply:
When a file is uploaded in application it gets uploaded alongside this Metadata: 
const metaData = {
        page: pageTitle,
        section: sectionTitle,
        componentName: key,
      };
all of which are COF specific, there is no other metadata we could use that would tell us that it was uploaded from a file upload filed from application for example. we're using this to only grab relevant files from the bucket rather than all files prefixed with the applicant's application_id.

Reply:
we could pass this in as a parameter with the default values specific to COF for now or add a TODO flag to let us know to change this for the next round/ fund? ideally we can add further metadata application side that we could use to filter these documents out assessment side?

Reply:
@gio-karunakaran can we filter by `file` key from the JSON e.g.
```           
             {
                "answer": "sample1.csv",
                "key": "ozgwXq",
                "title": "Risks to your project (document upload)",
                "type": "file"
              }
```
so filter the json by `"type": "file"`?

Reply:
Have refactored this to look into the json blob for the file name instead of looking in the S3 bucket as discussed 👍🏾